### PR TITLE
Delete hijacks functionality + frontend fixes (Closes #32 #131)

### LINKED ISSUE #32: Explicit user-controlled deletion of multiple hijacks
**Is your feature request related to a problem? Please describe.**
The user currently cannot delete hijacks using the UI

**Describe the solution you'd like**
This should be done after https://github.com/FORTH-ICS-INSPIRE/artemis/issues/34 is solved (will require hasura for "IN"-like searching in key lists).

UI button and backend db functionality to delete multiple hijacks and their related BGP updates.
Make sure that there is an "Are you sure?" banner before confirming final deletion.
This issue needs frontend support (UI) via graphql (delete hijacks and updates from db).
Delete only updates that belong to a single hijack, remove the key from the ones that belong to multiple (like withdrawals). Make sure that the UI is usable during the transition (e.g., you can delete first the hijack events and then the updates for safety).
We need a "delete" button within a hijack, and "delete" in the hijack table for multiple hijacks.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by pgigis**
### Description of PR

User can now delete a hijack from the hijacks view and hijack view. 

What component(s) does this PR affect?

- [X] Back-End (Database, Microservices, Containers, etc)
- [X] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Related Issue

Resolves #32 #131 

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: backend/core/database.py [hunk 1]
```diff
@@ -842,6 +858,31 @@ def handle_resolved_hijack(self, message):
...
                 log.exception("{}".format(raw))
 
+        def handle_delete_hijack(self, message):
+            raw = message.payload
+            log.debug("payload: {}".format(raw))
+            try:
+                redis_hijack_key = redis_key(
+                    raw["prefix"], raw["hijack_as"], raw["type"]
+                )
+                # if ongoing, force rekeying and delete persistent too
```
Comment (vkotronis) [inline backend/core/database.py]: remove this comment

File: frontend/webapp/views/main/templates/hijacks.htm [hunk 2]
```diff
@@ -515,34 +539,29 @@ <h1>Additional actions</h1>
...
                         display_multiple_actions += '</label></span>  <select class="form-control form-control-sm-auto" id="action_selection">';
-                        display_multiple_actions += '<option value="mark_resolved">Mark as Resolved</option>';
-                        display_multiple_actions += '<option value="mark_ignored">Mark as Ignored</option>';
-                        display_multiple_actions += '<option value="mark_seen">Mark as Acknowledged</option>';
-                        display_multiple_actions += '<option value="mark_not_seen">Mark as Not Acknowledged</option></select>';
+                        display_multiple_actions += '<option value="hijack_action_resolve">Mark as Resolved</option>';
+                        display_multiple_actions += '<option value="hijack_action_ignore">Mark as Ignored</option>';
+                        display_multiple_actions += '<option value="hijack_action_acknowledge">Mark as Acknowledged</option>';
+                        display_multiple_actions += '<option value="hijack_action_acknowledge">Mark as Not Acknowledged</option>';
```
Comment (vkotronis) [inline frontend/webapp/views/main/templates/hijacks.htm]: these two are the same???

File: frontend/webapp/core/actions.py [hunk 3]
```diff
@@ -12,122 +12,36 @@
...
-        self.connection = None
-        self.hijack_key = hijack_key
-        self.prefix = prefix
-        self.type_ = type_
-        self.hijack_as = hijack_as
-        self.init_conn()
-        self.hijack_exchange = Exchange(
-            "hijack-update", type="direct", durable=False, delivery_mode=1
+def rmq_hijack_action(obj):
+    log.info(
```
Comment (vkotronis) [inline frontend/webapp/core/actions.py]: this should be log.debug

File: frontend/webapp/core/actions.py [hunk 4]
```diff
@@ -12,122 +12,36 @@
...
+        "Send '{0}' hijack message with key: {1}".format(
+            obj["action"], obj["payload"]["key"]
         )
+    )
```
Comment (vkotronis) [inline frontend/webapp/core/actions.py]: you should also add an assertion here with what the obj should include (e.g., payload, routing_key, exchange, etc.)

File: frontend/webapp/core/actions.py [hunk 5]
```diff
@@ -12,122 +12,36 @@
...
-            log.exception("Learn_hijack_rule failed to connect to rabbitmq.")
-
     def on_response(self, message):
         if message.properties["correlation_id"] == self.correlation_id:
             self.response = message.payload
 
     def send(self, hijack_key, prefix, type_, hijack_as, action):
-        log.debug("sending")
+        log.debug(
+            "Send 'learn_new_rule - {0}' hijack message with key: {}".format(
```
Comment (vkotronis) [inline frontend/webapp/core/actions.py]: I think you mean key: {1}

File: frontend/webapp/static/js/custom/graphql.js [hunk 6]
```diff
@@ -262,57 +262,26 @@ function fetchDBVersion() { // eslint-disable-line no-unused-vars
     .catch(error => console.error(error));
 }
 
-
-function fetchLatestConfig() { // eslint-disable-line no-unused-vars
```
Comment (vkotronis) [inline frontend/webapp/static/js/custom/graphql.js]: has this been replaced with sth?

File: frontend/webapp/views/actions/actions_view.py [hunk 7]
```diff
@@ -31,61 +28,90 @@
...
-@actions.route("/hijacks/mitigate/", methods=["POST"])
-@roles_required("admin")
-def mitigate_hijack():
-    # log info
-    hijack_key = request.values.get("hijack_key")
-    prefix = request.values.get("prefix")
-
+def hijack_action():
+    data = json.loads(request.data.decode("utf-8"))
+    app.artemis_logger.info(data)
```
Comment (vkotronis) [inline frontend/webapp/views/actions/actions_view.py]: .debug

File: frontend/webapp/views/actions/actions_view.py [hunk 8]
```diff
@@ -31,61 +28,90 @@
...
     try:
-        _mitigate_hijack = Mitigate_hijack(hijack_key, prefix)
-        _mitigate_hijack.mitigate()
-    except BaseException:
-        app.artemis_logger.debug("mitigate_hijack failed")
-
-    return jsonify({"status": "success"})
-
+        action = data["action"]
```
Comment (vkotronis) [inline frontend/webapp/views/actions/actions_view.py]: add also assertions on what fields data should include

File: frontend/webapp/views/actions/actions_view.py [hunk 9]
```diff
@@ -253,25 +282,13 @@ def monitor_state():
...
-
-
-@actions.route("/hijacks_actions", methods=["POST"])
+@actions.route("/multiple_hijack_actions", methods=["POST"])
 @roles_required("admin")
 def submit_hijacks_actions():
-    hijack_keys = json.loads(request.values.get("hijack_keys"))
-    action = request.values.get("action")
+    data_ = json.loads(request.data.decode("utf-8"))
+    app.artemis_logger.info(data_)
```
Comment (vkotronis) [inline frontend/webapp/views/actions/actions_view.py]: .debug

File: frontend/webapp/views/main/templates/hijack.htm [hunk 10]
```diff
@@ -135,52 +136,42 @@ <h1 class="page-header">Viewing Hijack <small id="hijack_status"></small></h1>
...
-                                {% if mitigate == True %}
-                                <button id="mitigation_button" type="button" class="btn btn-danger" disabled="true">Mitigate</button>
-                                {% endif %}
-                            </div>
-                            <div class="col-lg-4">
-                                <button id="ignored_button" type="button" class="btn btn-warning" disabled="true">Ignore</button>
+                {% if current_user.has_roles('admin') %}
+                    <div class="panel panel-default">
+                        <div class="panel-heading">
+                            <i class="fa fa-bell fa-fw"></i> Hijack Actions
```
Comment (vkotronis) [inline frontend/webapp/views/main/templates/hijack.htm]: what are these fa, fw, etc.?

File: frontend/webapp/views/main/templates/hijack.htm [hunk 11]
```diff
@@ -742,240 +763,255 @@ <h5 class="modal-title" id="learnRuleModalLabel">Configuration diff (Learn new r
...
-                    success: function(result){
-                        if(result["status"] == "fail"){
-                            var message = '<div class="alert alert-danger alert-dismissible">'
-                            message += '<a href="#" class="close" data-dismiss="alert" aria-label="close">&times;</a>'
-                            message += result["response"] + '</div>'
-                            $('#comment_action').removeClass('btn-warning').addClass('btn-danger')
-                            $('#comment_action').html('Save');
+                        var message = [];
+                        if(result["status"] == "success"){
+                            $("#ignored_button").prop("disabled", true);
```
Comment (vkotronis) [inline frontend/webapp/views/main/templates/hijack.htm]: there is not ignored button anymore right?

File: frontend/webapp/views/main/templates/hijacks.htm [hunk 12]
```diff
@@ -660,16 +679,37 @@ <h1>Additional actions</h1>
...
-
-
-                {% endif %}
 
                 $('#timezone').html(display_timezone());
                 displayHelpTextButton();
                 $("#navbar_hijacks").addClass("active");
             });
 
+            {% if current_user.has_roles('admin') %}
```
Comment (vkotronis) [inline frontend/webapp/views/main/templates/hijacks.htm]: question: a user cannot simply change this client-side and be able to send queries via Flask?

File: frontend/webapp/views/main/templates/hijack.htm (hunk 10 above)
Comment (pgigis) [reply frontend/webapp/views/main/templates/hijack.htm]: Its reserved for icons using font-awesome

File: frontend/webapp/views/main/templates/hijacks.htm (hunk 12 above)
Comment (pgigis) [reply frontend/webapp/views/main/templates/hijacks.htm]: Even if he guess 'where' to send 'what', @roles_required("admin") in Flask should block the message.

File: frontend/webapp/views/actions/actions_view.py (hunk 8 above)
Comment (vkotronis) [reply frontend/webapp/views/actions/actions_view.py]: as discussed, no need for this

**Comment by vkotronis** [comment]
@vkotronis will also check if wiki updates are required after merge

**Pull request merged**